backend/amd64/emit.mlp

```diff
@@ -860,6 +860,20 @@ let emit_instr fallthrough i =
       I.movsxd (arg32 i 0) (res i 0)
   | Lop(Ispecific(Izextend32)) ->
       I.mov (arg32 i 0) (res32 i 0)
+  | Lop(Ispecific(Ibsr { arg_is_non_zero; })) ->
+    if arg_is_non_zero then begin
+      (* No need to handle that bsr is undefined on 0 input. *)
+      I.bsr (arg i 0) (res i 0);
+    end else begin
+      let lbl_z = new_label () in
+      let lbl_nz = new_label () in
+      I.bsr (arg i 0) (res i 0);
+      I.je (label lbl_z);
```

Comment:
If we're okay destroying the argument (or using a temporary)
```
I.bsr (arg i 0) (res i 0)
I.mov (int (-1)) (arg i 0)
I.cmov E (arg i 0) (res i 0)
```

Reply:
Nice trick using `cmov` from #172 and `mov` doesn't set the flags.  We can do it for clz and ctz with non-zero args too.

Reply:
Isn't there a problem if the argument and the result are the same register ?

Reply:
yes, it would need to use another register.